_includes/join_list_commas_and.html

```diff
@@ -0,0 +1,23 @@
+{% comment %}
+include as a function, which given a parameter `list`, will return the list joined with `,` but (also) using `and` between the last 2 elements.
+
+Uses include.list[0] to determine if the input is a list (should be falsey for a string or empty list) allowing individual strings to be passed in too.
+
+usage (without the extra spaces): 
+
+{ % include join_list_commas_and.html list=foo % }
+
+{% endcomment %}
+{% if include.list[0] %}
+    {% if include.list.size > 2 %}
+        {% assign last_item = include.list | last %}
+        {% assign new_last = "and " | append: last_item %}
+        {{ include.list | join: ", " | replace: last_item, new_last}}
```

Comment:
This does have slightly different output for lists of 2 items and 3+ items (matching existing behaviour):

`a and b` vs `a, b, and c`

If this is not desirable, the final comma can be excluded via the following patch: 

```suggestion
        {% assign list_len_minus_1 = include.list.size | minus: 1 %}
        {% assign list_excluding_last = include.list | slice: 0, list_len_minus_1 %}
        {{ list_excluding_last | join: ", " }} and {{ last_item }}
```

which would then produce `a, b and c`.

I'll leave this up to you @ubdbra001

Reply:
I'm happy with the original, so will keep it as is.